pyproject.toml

```diff
@@ -0,0 +1,132 @@
+maketx = "OpenImageIO:_command_line"
+oiiotool = "OpenImageIO:_command_line"
+
+[build-system]
+build-backend = "scikit_build_core.build"
+requires = [
+    "scikit-build-core>=0.10.6,<1",
+    "pybind11",
+]
+
+[tool.scikit-build]
+build.verbose = true
+# Exclude unnecessary directories from the source distribution.
+sdist.exclude = [".github", "testsuite", "ASWF", "docs"]
+# Pin minimum scikit-build-core to that specified in build-system.requires.
+minimum-version = "build-system.requires"
+# Pin minimum CMake version to that specified in CMakeLists.txt.
+cmake.version = "CMakeLists.txt"
+wheel.license-files = ["LICENSE.md", "THIRD-PARTY.md"]
+# Make sure the package is structured as expected.
+wheel.install-dir = "OpenImageIO"
+# Only install the user and fonts components.
+install.components = ["user", "fonts"]
+
+[tool.scikit-build.cmake.define]
+# Build missing dependencies. See src/cmake for details.
+OpenImageIO_BUILD_MISSING_DEPS = "all"
+# Don't build tests. Dramatically improves build time.
+OIIO_BUILD_TESTS = "0"
+# Prefer linking static dependencies when possible.
+LINKSTATIC = "1"
+# Standardize the install directory for libraries, as expected by
+# other parts of the wheels build process.
+CMAKE_INSTALL_LIBDIR = "lib"
+
+# Dynamically set the package version metadata by pasrsing CMakeLists.txt.
+[tool.scikit-build.metadata.version]
+provider = "scikit_build_core.metadata.regex"
+input = "CMakeLists.txt" 
+regex = 'set \(OpenImageIO_VERSION "(?P<value>[0-9a-z.]+)"\)'
+
+
+# On macOS, ensure dependencies are only built for the target architecture.
+[[tool.scikit-build.overrides]]
+if.platform-system = "darwin"
+if.platform-machine  = "arm64"
+inherit.cmake.define = "append"
+cmake.define.CMAKE_OSX_ARCHITECTURES = "arm64"
+
+[[tool.scikit-build.overrides]]
+if.platform-system = "darwin"
+if.platform-machine  = "x86_64"
+inherit.cmake.define = "append"
+cmake.define.CMAKE_OSX_ARCHITECTURES = "x86_64"
+
+
+[tool.cibuildwheel]
+build-verbosity = 1
+skip = [
+    # Skip 32-bit builds
+    "*-win32",
+    "*-manylinux_i686",
+    # Building with musl seems to work, but the repair-wheel step seems to fail...
+    # This may be a bug in repairwheel (or auditwheel)?
+    "*musllinux*",  
+]
+test-command = "oiiotool --buildinfo"
+# Prevent CIBW's default repair-wheel command from running, because it fails to
+# recognize that the wheel isn't broken to begin with, and ends up creating
+# additional copies of the shared libraries in the wheel.
+
+[tool.cibuildwheel.macos.environment]
+# Specify a persistent build directory so the repairwheel command can (re)find
+# dependency libraries built by OpenImageIO.
+SKBUILD_BUILD_DIR = "/tmp/build_oiio_wheels"
+# Always (re)build the TIFF library; see #4439 for more details.
+SKBUILD_CMAKE_ARGS = "-DOpenImageIO_BUILD_LOCAL_DEPS=TIFF; -DOpenImageIO_BUILD_MISSING_DEPS=all; -DLINKSTATIC=1; -DIGNORE_HOMEBREWED_DEPS=1; -DCMAKE_INSTALL_LIBDIR=lib"
```

Comment:
Is it necessary to pass build missing here since it's already defined at the top level?

Reply:
Good call. 

I'm also removing stuff related to incremental building with cibuildwheel. I've come to realize that `checked_find_package` installs libraries after it builds 'em; but successive OIIO builds that are able to find previously-built libraries in <build-dir>/deps/dist never have the opportunity to be installed (cuz their build recipes are never invoked). This is how and why the custom repair-wheel-cmd worked -- `repairwheel -l $SKBUILD_BUILD_DIR/deps/dist ...` finds and effectively "installs" any shared libraries that `checked_find_package` successfully bundled into the cpython-3.8 wheel, but failed to bundle into the cpython-3.9+ wheels. 

In any case, it's a problem outside the scope of this PR, I think.